src/scratch/ScratchRuntime.as

```diff
@@ -89,49 +145,342 @@ public class ScratchRuntime {
 		processEdgeTriggeredHats();
 		interp.stepThreads();
 		app.stagePane.commitPenStrokes();
+		
+		if (ready>=0) {
+			app.stagePane.countdown(count);
+		}
 	}
 
-//-------- recording test ---------
+//-------- recording video code ---------
 	public var recording:Boolean;
 	private var frames:Array = [];
-
+	private var sounds:Array = [];
+	private var recTimer:Timer;
+	private var baFlvEncoder:ByteArrayFlvEncoder;
+	private var position:int;
+	private var secs:Number;
+	private var alreadyDone:int;
+	
+	private var pSound:Boolean;
+	private var mSound:Boolean;
+	private var mouse:Boolean;
+	private var cursor:Boolean;
+	public var fullEditor:Boolean;
+	private var framerate:Number;
+	private var width:int;
+	private var height:int;
+	public var ready:int=-1;
+	
+	private var mBytes:ByteArray;
+	private var mPosition:int = 0;
+	private var mic:Microphone;
+	private var mReady:Boolean;
+	
+	private var timeout:int;
+	
 	private function saveFrame():void {
-		var f:BitmapData = new BitmapData(480, 360);
-		f.draw(app.stagePane);
-		frames.push(f);
-		if ((frames.length % 100) == 0) {
-			trace('frames: ' + frames.length + ' mem: ' + System.totalMemory);
+		saveSound();
+		var t:Number = getTimer()*.001-secs;
+		while (t>sounds.length/framerate+1/framerate) {
+			saveSound();
+		}
+		if (cursor) var cursor:DisplayObject = Resources.createDO('videoCursor');
+		if (mouse && app.gh.mouseIsDown) var circle:Bitmap = Resources.createBmp('mouseCircle');
+		var f:BitmapData;
```

Comment:
It may be possible to reduce garbage collection overhead by reusing the same `BitmapData` across multiple frames as long as the width and height don't change. I'm not sure if that would be worth the extra complexity or not... just a thought.